### PR TITLE
BUG: Added code for missing listify input of []

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses #830

Improves code coverage of `listify`. Now works on empty lists.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. Please also list any relevant details for
your test configuration

- Expanded range of inputs of existing tests to include `[]` and `[[]]` inputs.

**Test Configuration**:
* Operating system: MacOS
* Version number: Python 3.9
* Any details about your local setup that are relevant

# Checklist:

- [ ] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by rstoneback** [comment]
Getting a failure since the pandas version installed is no longer compatible with numpy 1.17, needs minimum of 1.17.3

**Comment by rstoneback** [comment]
Currently getting a JSON error which may be related to F10.7 downloads. Also errors are popping up on a custom function check which calculates 'double_mlt'. This is on the 1.17.3 version.

**Comment by rstoneback** [comment]
Failures in 3.9 are related to F10.7 downloads. 3.8 seems to be getting by ok for some reason.

**Comment by rstoneback** [comment]
Better breakdown of the issue in 3.7
```
>       assert np.all(self.testInst[fixed, 'doubleMLT']
177
                      == 2. * self.testInst[fixed, 'mlt'])
178
E       assert False
179
E        +  where False = <function all at 0x7fed810733b0>(Epoch\n2009-01...dtype: float64 == Epoch\n2009-01...dtype: float64
180
E        +    where <function all at 0x7fed810733b0> = np.all
181
E           Use -v to get the full diff)
182
```

**Comment by rstoneback** [comment]
Added issue for pandas version, https://github.com/pysat/pysat/issues/833

**Comment by rstoneback** [comment]
Logging numpy as well.

https://github.com/pysat/pysat/issues/834

File: pysat/utils/_core.py [hunk 1]
```diff
@@ -136,6 +136,8 @@ def listify(iterable):
         list_iter = [arr_iter.tolist()]
     elif arr_iter.shape[0] >= 1:
         list_iter = arr_iter.flatten().tolist()
+    elif arr_iter.shape[0] == 0:
+        list_iter = arr_iter.tolist()
```
Comment (aburrell) [inline pysat/utils/_core.py:140]: Is there any other condition that could be triggered here?  If not, the `elif` above should just be `else`.  If there is, let's add an error check here that is triggered for `else`.

Comment (rstoneback) [reply pysat/utils/_core.py:138]: Updated.

**Comment by jklenzing** [comment]
> Getting a failure since the pandas version installed is no longer compatible with numpy 1.17, needs minimum of 1.17.3

On July 26, 1.18 becomes the new minimum version => https://numpy.org/neps/nep-0029-deprecation_policy.html

**Review by aburrell — APPROVED** [review]
🐴  Looking good!

**Pull request merged**